extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -59,21 +63,21 @@ public class DefaultTenantConfigResolver {
     @Inject
     Instance<UserInfoCache> userInfoCache;
 
-    @Inject
-    Event<SecurityEvent> securityEvent;
-
     @Inject
     @ConfigProperty(name = "quarkus.http.proxy.enable-forwarded-prefix")
     boolean enableHttpForwardedPrefix;
 
     private final BlockingTaskRunner<OidcTenantConfig> blockingRequestContext;
+    private final boolean securityEventObserved;
+    private final Event<SecurityEvent> securityEvent;
+    private final ConcurrentHashMap<String, BackChannelLogoutTokenCache> backChannelLogoutTokens = new ConcurrentHashMap<>();
 
-    private volatile boolean securityEventObserved;
-
-    private ConcurrentHashMap<String, BackChannelLogoutTokenCache> backChannelLogoutTokens = new ConcurrentHashMap<>();
-
-    public DefaultTenantConfigResolver(BlockingSecurityExecutor blockingExecutor) {
+    public DefaultTenantConfigResolver(BlockingSecurityExecutor blockingExecutor, Event<SecurityEvent> securityEvent,
+            BeanManager beanManager, SecurityConfig securityConfig) {
         this.blockingRequestContext = new BlockingTaskRunner<OidcTenantConfig>(blockingExecutor);
+        this.securityEventObserved = SecurityEventHelper.eventObserved(new SecurityEvent(null, (SecurityIdentity) null),
```

Comment:
Should it be `SecurityEventHelper.eventObserved(SecurityEvent.class)` ?

Reply:
This goes down to the `jakarta.enterprise.inject.spi.BeanManager#resolveObserverMethods`, I'm just spec folk had some reason for this, but I don't know why looking at `io.quarkus.arc.impl.BeanManagerImpl#resolveObserverMethods`  it just do `event.getClass`. Anyway, I can't change this.

Reply:
I agree @sberyozkin but I won't touch Arc project in this PR because I don't know why things are done there as they are done. CDI experts could say it should be added to spec first. It is not important enough, but if CDI experts agree, we can do it.

Reply:
So IMHO not possible.

Reply:
@michalvavrik But if observing events with GRPC won't work then it can be in Vertx HTTP ? In any case, indeed, not a big problem, it is an internal code, so can be optimized eventually, thanks